fluss-server/src/main/java/com/alibaba/fluss/server/replica/fetcher/RemoteLeaderEndpoint.java

```diff
@@ -86,7 +94,36 @@ public CompletableFuture<Map<TableBucket, FetchLogResultForBucket>> fetchLog(
             FetchLogRequest fetchLogRequest) {
         return tabletServerGateway
                 .fetchLog(fetchLogRequest)
-                .thenApply(RpcMessageUtils::getFetchLogResult);
+                .thenApply(
+                        fetchLogResponse -> {
+                            Map<TableBucket, FetchLogResultForBucket> fetchLogResultMap =
+                                    new HashMap<>();
+                            List<PbFetchLogRespForTable> tablesRespList =
+                                    fetchLogResponse.getTablesRespsList();
+                            for (PbFetchLogRespForTable tableResp : tablesRespList) {
+                                long tableId = tableResp.getTableId();
+                                List<PbFetchLogRespForBucket> bucketsRespList =
+                                        tableResp.getBucketsRespsList();
+                                for (PbFetchLogRespForBucket bucketResp : bucketsRespList) {
+                                    TableBucket tableBucket =
+                                            new TableBucket(
+                                                    tableId,
+                                                    bucketResp.hasPartitionId()
+                                                            ? bucketResp.getPartitionId()
+                                                            : null,
+                                                    bucketResp.getBucketId());
+                                    Replica replica =
+                                            replicaManager.getReplicaOrException(tableBucket);
```

Comment:
According to the **"Principle of Least Knowledge"** (also known as the Law of Demeter), `replicaManager` represents an unnecessarily large dependency in this context, as we only need to retrieve the table path. A cleaner approach would be to add the `TablePath` to both `BucketFetchStatus` and `InitialFetchStatus`. 

In the `buildFetchLogRequest` method, we can construct a `Map<Long, TablePath>` from the existing `Map<TableBucket, BucketFetchStatus> replicas`. This map can then be passed to `LeaderEndpoint#fetchLog`, ensuring that the dependency remains minimal and the code adheres to the principle of least knowledge.

Reply:
Done!